network/Message.cpp

```diff
@@ -140,41 +115,29 @@ void swap(Message& lhs, Message& rhs)
 
 void BufferToHeader(const Message::HeaderBuffer& buffer, Message& message) {
     message.m_type = static_cast<Message::MessageType>(buffer[0]);
-    message.m_sending_player = buffer[1];
-    message.m_receiving_player = buffer[2];
-    message.m_synchronous_response = (buffer[3] != 0);
-    message.m_message_size = buffer[4];
+    message.m_synchronous_response = (buffer[1] != 0);
+    message.m_message_size = buffer[2];
 }
 
 void HeaderToBuffer(const Message& message, Message::HeaderBuffer& buffer) {
     buffer[0] = message.Type();
-    buffer[1] = message.SendingPlayer();
-    buffer[2] = message.ReceivingPlayer();
-    buffer[3] = message.SynchronousResponse();
-    buffer[4] = message.Size();
+    buffer[1] = message.SynchronousResponse();
+    buffer[2] = message.Size();
```

Comment:
Could you replace these magic constants with an enumeration and then use the enumeration in place of the constants?
````
class Message {
public:
...
enum class Parts {TYPE, RESPONSE, SIZE};
}
````

I know that the old code uses the magic numbers, but this would improve things.